File: src/main/java/com/kaala/java/prime/Prime.java
package com.kaala.java.prime;

import java.util.stream.IntStream;

/**
 * @author Binod Shrestha <binodshrestha@lftechnology.com> on 5/18/17.
 */
public class Prime {

    private static boolean isPrimeNaive(Integer num) {
        for (int i = 2; i < num / 2; i++) {
            if (num % i == 0) {
                return false;
            }
        }

        return true;
    }

    private static boolean isPrimeNaiveStream(final Integer num) {
        return IntStream.range(2, num/2).noneMatch(i -> num%i == 0);
    }

    public static void main(String[] args) {
        System.out.println(isPrimeNaiveStream(7));
    }

}

File: src/main/java/com/kaala/kotlin/prime/Prime.kt
package com.kaala.kotlin.prime

/**
 * @author Binod Shrestha <binodshrestha@lftechnology.com> on 5/27/17.
 */

fun isPrimeNaive(num: Int): Boolean {
    return (2..num / 2).none { n ->  num % n == 0 }
}

fun main(args: Array<String>) {
    val start = System.nanoTime()
    println(isPrimeNaive(7))
    val elapsedTime = System.nanoTime() - start

    println(elapsedTime)

}